.gitignore

```diff
@@ -36,6 +36,7 @@ README.html
 *.swp
 *.autosave
 *~
+compile_commands.json
```

Comment:
Added by vscode?

Reply:
After setting a CMake flag, it will be automatically generated during build. It is used by vscode for non-buck cpp project.